src/drivers/imx/sai.c

```diff
@@ -366,7 +375,12 @@ static inline int sai_set_config(struct dai *dai, struct ipc_config_dai *common_
 		break;
 	}
 
-	val_cr4 |= REG_SAI_CR4_FRSZ(sai->params.tdm_slots);
+	if (tdm_mode == TDM_ENABLE)
+		val_cr4 |= REG_SAI_CR4_FRSZ(sai->params.tdm_slots);
+	else
+		val_cr4 |= REG_SAI_CR4_FRSZ(
+		    (sai->params.tdm_slots == 1) ? 2 : sai->params.tdm_slots);
```

Comment:
This is more restricted than the "old" configuration.
Meaning, before these changes we were setting 
`val_cr4 |= REG_SAI_CR4_FRSZ(sai->params.tdm_slots);`
for all formats and platforms.

There is no possibility to have 1 channel (or 1 word per frame) in other case than DSP-A or DSP-B format?

Reply:
I think the reason is as Daniel said, we need to use 2 words even on mono, because the "wire" from DAI to codec uses I2S protocol! which has a fixed number of 2 channels. @dbaluta ,Please correct me if I was wrong.

Reply:
this is a convention we need to make between SAI implementation  and setting the format in topology.

We sai like this: we will implement true TDM with dsp_a/dsp_b and true I2S with the rest of the formats. This is not ideal but is best we can do with little effort.

Reply:
Please add a comment with this explanation, in code, and make tdm_mode bool as suggested here https://github.com/thesofproject/sof/pull/8200/files#r1326674287